lms/product/plugin/misc.py

```diff
@@ -61,3 +61,27 @@ def get_document_url(self, request):
             )
 
         return assignment.document_url if assignment else None
+
+    def get_deeplinking_launch_url(self, request, _assignment_configuration: dict):
+        """
+        Launch URL for deep linked assignments.
+
+        To which URL to point the deep linked assignments in the LMS.
+        This URL wel'll be part of the deep link response message
+        to finish the configuration during a deep link request.
+        """
+        # In general we'll point to our regular basic launch URL.
+        # The assignment configuration we'll be retrieved by other methods (eg, custom parameters) so that parameter is not used here.
+        return request.route_url("lti_launches")
+
+    def get_deep_linked_assignment_configuration(self, request):
+        """Get the configuration of an assignment that was original deep linked."""
+        params = {}
+        possible_parameters = ["url", "group_set"]
+
+        for param in possible_parameters:
+            # Get the value from the custom parameters set during deep linking
+            if value := request.lti_params.get(f"custom_{param}"):
+                params[param] = value
+
+        return params
```

Comment:
I feel like this is premature. We only have two examples and they are totally different?

I think the assumption here is that D2L is doing it right.